core/src/test/java/org/apache/calcite/test/RexProgramTest.java

```diff
@@ -624,6 +624,13 @@ private RexProgramBuilder createProg(int variant) {
             case_(isNotNull(i0), i0, i1), c01),
         is(true));
 
+    final RelDataType intArray = typeFactory.createArrayType(intType, -1);
+    RexInputRef inputRef = rexBuilder.makeInputRef(intArray, 0);
+    RexNode rexNode = rexBuilder.makeCall(
+        SqlStdOperatorTable.ITEM,
+        inputRef,
+        rexBuilder.makeLiteral(0, intType, false));
```

Comment:
We may also need to add some tests for the `IS NULL/IS NOT NULL` simplification.

Reply:
How about make it a separate test ?

Reply:
Added this for IS NULL.  I did not see IS NOT NULL API in Strong.

Reply:
Can we also add some test cases in `#simplifyStrong` ? Also, you can create a tool method to create the item call, just like `plus(` and `sub(`.

Reply:
I created a utility `RexProgramBuilder.item(RexInputRef inputRef, int index)` and use it in the unit test.  Regarding `simplifyStrong` it seems it will do essentially the same test that `isItemStrong` is doing since ITEM is merely an accessor, not an evaluation function. Let me know if you have a specific test in mind that is different from `isItemStrong`.

Reply:
A test case such as 'a.item(b+c) is null'.

Reply:
I think it was meant that `a` and `b` are numeric columns, so `a.item(b+c)` is a call for obtaining array element, but I think that this case is not connected with `ITEM` itself since it checks how strong logic is applied to function arguments, and we already have such tests.

Reply:
Yes, we have that, i just want to see a test that the composite of `ITEM` and other `STRONG` operators can be simplified correctly.